### DISCUSSION TITLE
XML to JSON

### DISCUSSION THREAD

**Discussion by LanciaH** [nextgenhealthcare/connect#5178] [Q&A] [answered]
Hi Me Again,
Version Mirth Connect Server 4.0.1.

I created a channel where the source is a Database Reader into an oracle database.
Simple sql statement that transforms into XML.  This part is working correctly/

I have to send this data in JSON format via HTTP, 

My Question is where do I do the conversion from XML to JSON?  I see other post mentioning code, but where is the code placed?

The vendor provided the JSON fields.
Thanks

**Comment by pacmano1** [comment]
In a transformer step of type Javascript in general.  You can find videos on YouTube that do walkthroughs.   While not your data type conversion, see https://www.youtube.com/watch?v=0cuhCL9xKmg.  There are others videos for other data type conversions.

**Reply by pacmano1** [reply]
Also, Oracle like other databases can generate native JSON.   While the database reader will wrap that JSON in XML in a single XML tag, you can "unwrap" it in the first step of your transformer.

**Comment by LanciaH** [comment]
Great idea about using Oracle to create the JSON, let me investigate that option.

Thanks!

**Comment by LanciaH** [comment, marked as answer]
I used a Transformer to convert the XML into JSON, so far so good.
Thanks all

```javascript
var jsEntity = GetEntities();
  
  var jsMedications = GetMedications();
  jsMedications.name = msg['name'].toString();
  jsMedications.parentid = msg['parentid'].toString();
  jsMedications.externalid = msg['externalid'].toString();
  
 

  jsEntity.entities.push(jsMedications);  // ADD IT TO THE ENTITY ARRAY
  channelMap.put('Entity_Locations',JSON.stringify(jsEntity));

  
  
//=============================
function GetEntities()
{
  var jsData = 
	{
	  entities: []
	};

  return jsData;
}

//===========================
function GetMedications()
{
  var jsData = 
	{
	 name: "",
      parentid: "",
      externalid: "",
	};

  return jsData;
}
```

**Reply by pacmano1** [reply]
Reminder to use code highlighting here:
```JS
var jsEntity = GetEntities();

var jsMedications = GetMedications();
jsMedications.name = msg['name'].toString();
jsMedications.parentid = msg['parentid'].toString();
jsMedications.externalid = msg['externalid'].toString();

jsEntity.entities.push(jsMedications); // ADD IT TO THE ENTITY ARRAY
channelMap.put('Entity_Locations', JSON.stringify(jsEntity));

//=============================
function GetEntities() {
    var jsData = {
        entities: []
    };

    return jsData;
}

//===========================
function GetMedications() {
    var jsData = {
        name: "",
        parentid: "",
        externalid: "",
    };

    return jsData;
}
```

**Reply by LanciaH** [reply]
Sorry but how do you do code highlighting?  Thanks

**Reply by FTK7** [reply]
Select the code and mark it as code, in the editor options:
![imatge](https://user-images.githubusercontent.com/66680696/169035792-eeebd798-2ce9-45e3-9e5b-1085c0847bea.png)

**Reply by LanciaH** [reply]
Did that but how do you get it to use the colors?  I tried ...JSON and all the text turns red

**Reply by FTK7** [reply]
You can specify the language after de backticks: 

````
```javascript
code
```
````

**Reply by LanciaH** [reply]
got it, Thanks!
